feat: Add Flutter 1.0 launch article (#7997)
supabase/supabase: apps/www/_blog/2022-08-01-sharing-with-you-the-supabase-flutter-sdk-1-developer-preview.mdx

---
title: 'Sharing With You the Supabase Flutter SDK 1.0 Developer Preview'
description: Supabase Flutter SDK is getting a major update and we need your help making it better.
author: tyler_shukert
image: flutter-1/supabase-flutter-1.jpg
thumb: flutter-1/supabase-flutter-1.jpg
tags:
  - flutter
  - mobile
date: '2022-08-01'
toc_depth: 3
---

Today, we are announcing release of the developer preview version of v1.0 of [Supabase Flutter SDK](https://pub.dev/packages/supabase_flutter). Flutter has quickly become one of the most popular frameworks for developers that want to build cross-platform mobile apps. We can attest to that growth, our Flutter SDK is one of the most popular ones and each day we see more and more Flutter devs choosing Supabase to handle the backend.

For this release, our main focus was developer experience, so you will see several improvements in that area alongside new features and enhancements. Because we want to provide the best developer experience possible, we would love it if you could try our new SDK and give us feedback. 

> Before we dive into the actual updates though, I would like to thank all the community contributors who have helped the library to be where it is today. 

Author: @kiwicopple Were you thinking to list every single contributors including all the sub-component libraries as well?

final data = response.data;

// After
final data = await Supabase.instance.from('messages').select();
```

## Desktop support for deeplinks

Ever since supabase-flutter was born, it supported only iOS, Android and Web for deep linking. This was a limitation of the deep link library that we were using, but with the 1.0 launch, we are moving to use [app_links](https://pub.dev/packages/app_links), which will enable us to support MacOS and Windows applications as well! Linux support is being worked on as well. 

![Supabase Flutter desktop support](/images/blog/flutter-1/supported-platforms-table.png)

## Multiplayer support

[Multiplayer](https://supabase.com/blog/2022/04/01/supabase-realtime-with-multiplayer-features) is the next generation Supabase realtime engine that was announced at the previous launch week as coming out soon. We want our Flutter developers to experience this new multiplayer feature as well, so are working hard at bringing it to our Flutter SDK. It is not yet included in the developer preview of Supabase Flutter 1.0, but will be part of it when stable launch has been released. 

## Supabase Auth UI for Supabase

> Last but not least, we are bringing you another library, the Supabase Auth UI for Supabase! When released, this library will enable you to implement a basic authentication screen without having to build it yourself. You can just load the library and display a nice looking auth ui. The library will take your theme settings automatically to match the look and feel of your application. You can easily enable/ disable certain auth features with the configuration as well. I would like to thank [Fatuma](https://twitter.com/XquisiteDreamer) for single handedly working on bringing us an easier authentication experience. 

Reviewer: ```suggestion
Last but not least, we are bringing you another library, the Supabase Auth UI for Supabase! When released, this library will enable you to implement a basic authentication screen without building it yourself. You can just load the library and display a nice looking Auth UI. The library takes your theme settings automatically to match the look and feel of your application. You can easily enable/disable certain Auth features with the configuration.

I would like to thank [Fatuma](https://twitter.com/XquisiteDreamer) for single-handedly working on bringing us an easier authentication experience. 
```

## Final thoughts

> The updates that I got to introduce here are just the tip of the iceberg, and there have been many bug fixes or other features constantly being added to the Supabase Flutter SDK. This could not have been possible with the help from the open source community. Here, I would also like to give a shout out to two other developers who have been a major part of the journey of this SDK, and they are [Vinzent](https://twitter.com/Vinzent03_) and [Daniel Mossaband](https://github.com/DanMossa). They have been a huge part of the Supabase Flutter SDK not just for the 1.0 release, but throughout the lifetime of the library. 

Reviewer: ```suggestion
These updates are just the tip of the iceberg for 1.0. There are been many bug fixes and features constantly being added to the Supabase Flutter SDK. This could not have been possible with the help from the open source community. Here, I would also like to give a shout out to two other developers who have been a major part of the journey of this SDK: [Vinzent](https://twitter.com/Vinzent03_) and [Daniel Mossaband](https://github.com/DanMossa). They have been a huge part of the Supabase Flutter SDK - not just for the 1.0 release, but throughout the lifetime of the library. 
```
